### PR TITLE
More code updates based on code review with Tim/Abhipsa

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Messaging/IncomingMessage.cs [hunk 1]
```diff
@@ -109,24 +109,6 @@ public uint DeliveryCount
...
-        /// <remarks>
-        /// Possible values:
-        /// <list type="bullet">
-        /// <item>none (default): no feedback message is generated.</item>
-        /// <item>positive: receive a feedback message if the message was completed.</item>
-        /// <item>negative: receive a feedback message if the message expired (or maximum delivery count was reached) without being completed by the device.</item>
-        /// <item>full: both positive and negative.</item>
-        /// </list>
-        /// </remarks>
-        public string Acknowledgement
```
Comment (drwill-ms) [inline iothub/device/src/Messaging/IncomingMessage.cs:124]: Not used on the device-side.

File: iothub/device/src/Pipeline/TransportHandlerFactory.cs [hunk 2]
```diff
@@ -27,13 +27,6 @@ public IDelegatingHandler Create(PipelineContext context)
                     iotHubClientMqttSettings);
             }
 
-            if (transportSettings is IotHubClientHttpSettings iotHubClientHttpSettings)
```
Comment (drwill-ms) [inline iothub/device/src/Pipeline/TransportHandlerFactory.cs:30]: Http not allowed in pipeline as it won't be a main device transport.

File: iothub/device/src/Messaging/IncomingMessage.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// The data structure that represents the message that is received from IotHub.
+    /// The data structure that represents the message that is received from Iot hub.
```
Comment (brycewang-microsoft) [inline iothub/device/src/Messaging/IncomingMessage.cs:10]:
```suggestion
    /// The data structure that represents the message received from IoT Hub.
```

File: iothub/device/src/Messaging/OutgoingMessage.cs [hunk 4]
```diff
@@ -73,16 +76,16 @@ public DateTime CreationTimeUtc
...
-            set => SystemProperties[MessageSystemPropertyNames.CreationTimeUtc] = value;
+            get => GetSystemProperty<DateTime>(MessageSystemPropertyNames.CreationTimeBatchUtc);
+            set => SystemProperties[MessageSystemPropertyNames.CreationTimeBatchUtc] = value;
         }
 
         /// <summary>
         /// The time when this message is considered expired
         /// </summary>
-        public DateTime ExpiryTimeUtc
+        public DateTime ExpiresOnUtc
```
Comment (brycewang-microsoft) [inline iothub/device/src/Messaging/OutgoingMessage.cs:88]: Align with naming pattern of other properties?

```suggestion
        public DateTime ExpiredOnUtc
```

Comment (timtay-microsoft) [reply iothub/device/src/Messaging/OutgoingMessage.cs:88]: This field is in the future, unlike the other date times in here

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/TransportSettings/AmqpConnectionPoolSettings.cs [hunk 5]
```diff
@@ -22,6 +20,9 @@ public sealed class AmqpConnectionPoolSettings
         /// </remarks>
         private const uint DefaultPoolSize = 100;
 
+        // Iot hub allows up to 999 tokens per connection. Setting the threshold just below that.
+        internal const uint MaxDevicesPerConnection = 995;
```
Comment (brycewang-microsoft) [inline iothub/device/src/TransportSettings/AmqpConnectionPoolSettings.cs:24]: I am curious why not 998 here? Or maybe we don't care much about this?

**Review by brycewang-microsoft — APPROVED** [review]

File: iothub/device/src/Messaging/OutgoingMessage.cs (hunk 4 above)
Comment (drwill-ms) [reply iothub/device/src/Messaging/OutgoingMessage.cs:88]: Expiration date is more likely in the future, so we opted for present/future tense.

File: iothub/device/src/TransportSettings/AmqpConnectionPoolSettings.cs (hunk 5 above)
Comment (drwill-ms) [reply iothub/device/src/TransportSettings/AmqpConnectionPoolSettings.cs:24]: Dunno what the historical context was.

**Pull request merged**